Add Revision Support Feature Documentation

tharikaGitHub opened the pull request:
## Purpose
This PR adds documentation for the Revision Support Feature. Please find the document screenshots below.

<img width="1463" alt="1" src="https://user-images.githubusercontent.com/8557410/105859134-2f0d2300-6012-11eb-8541-93143fa208b4.png">
<img width="1385" alt="2" src="https://user-images.githubusercontent.com/8557410/105859151-35030400-6012-11eb-82e5-7f552c1dd81a.png">
<img width="1368" alt="3" src="https://user-images.githubusercontent.com/8557410/105859170-37fdf480-6012-11eb-8014-1b5ce012ec0e.png">
<img width="1396" alt="4" src="https://user-images.githubusercontent.com/8557410/105859179-3a604e80-6012-11eb-8175-b3aada0c32f8.png">
<img width="1377" alt="5" src="https://user-images.githubusercontent.com/8557410/105859189-3c2a1200-6012-11eb-84fe-dac3b2c2ab58.png">
<img width="1355" alt="6" src="https://user-images.githubusercontent.com/8557410/105859195-3df3d580-6012-11eb-93bb-16eb9c399e37.png">
<img width="1372" alt="7" src="https://user-images.githubusercontent.com/8557410/105859201-3fbd9900-6012-11eb-9946-0856420a9c0b.png">
<img width="1358" alt="8" src="https://user-images.githubusercontent.com/8557410/105859208-41875c80-6012-11eb-8b03-0e4b71358f3e.png">

tmkasun commented:
I think some of these screenshots will change after addressing the code review suggestion we got during the last review session, For example, Show an icon inside the review circles indicating whether it is deployed somewhere (to an environment) or not

tharikaGitHub commented:
> I think some of these screenshots will change after addressing the code review suggestion we got during the last review session, For example, Show an icon inside the review circles indicating whether it is deployed somewhere (to an environment) or not

Yes @tmkasun this is to add documentation for the milestone release so the users can try it out with the existing UI. We will change the screenshots after addressing the review concerns.

Mariangela commented on en/docs/design/create-api/create-api-revisions.md:
```diff
@@ -0,0 +1,89 @@
+
+## Restore a revision
+
+If you need to restore a specific revision, click on the **Restore** button in the revision list as shown below. This will restore the revision in to the working copy.
```
Is the terminology "working copy" correct here? Or has a new name been decided on?

tharikaGitHub replied:
Yes this should be changed.

tharikaGitHub replied:
Shall we change it to ```This will restore the API with the specific revision.```?

Mariangela replied:
@tharikaGitHub  +1 to the suggested change
